monai/transforms/spatial/array.py

```diff
@@ -2682,39 +2679,104 @@ def one_fn(patch):
     def threshold_fn(self, patch):
         return patch.sum() < self.threshold
 
-    def __call__(self, array: NdarrayOrTensor):
-        # create the patch iterator which sweeps the image row-by-row
-        array_np, *_ = convert_data_type(array, np.ndarray)
-        patch_iterator = iter_patch(
-            array_np,
-            patch_size=(None,) + self.patch_size,  # expand to have the channel dim
-            start_pos=(0,) + self.offset,  # expand to have the channel dim
-            overlap=self.overlap,
-            copy_back=False,
-            mode=self.pad_mode,
-            **self.pad_kwargs,
-        )
-
-        if self.sort_fn is not None:
-            patch_iterator = sorted(patch_iterator, key=self.sort_fn)
-
-        output = [
-            (convert_to_dst_type(src=patch, dst=array)[0], convert_to_dst_type(src=slices[..., 0], dst=array)[0])
-            for patch, slices in patch_iterator
-            if self.filter_fn(patch)
-        ]
-
+    def __call__(self, image: NdarrayOrTensor):
+        # Create the patches that sweeps the image row-by-row
+        patched_image, locations = self.create_patches(image)
+        # Filter patches
         if self.num_patches:
-            output = output[: self.num_patches]
-            if len(output) < self.num_patches:
-                patch = convert_to_dst_type(
-                    src=np.full((array.shape[0], *self.patch_size), self.pad_kwargs.get("constant_values", 0)),
-                    dst=array,
-                )[0]
-                start_location = convert_to_dst_type(src=np.zeros((len(self.patch_size), 1)), dst=array)[0]
-                output += [(patch, start_location)] * (self.num_patches - len(output))
+            patched_image, locations = self.filter_count(patched_image, locations)
+        elif self.threshold:
+            patched_image, locations = self.filter_threshold(patched_image, locations)
 
-        return output
+        # Create constant value patches if needed.
+        if self.num_patches:
+            if len(patched_image) < self.num_patches:
+                n_extras = self.num_patches - len(patched_image)
+                padding = [(0, n_extras)] + [(0, 0)] * len(image.shape)
+                patched_image = np.pad(patched_image, padding, mode="constant", **self.pad_kwargs)
+                if self.return_location:
+                    padding = [(0, n_extras), (0, 0)]
+                    locations = np.pad(locations, padding, mode="constant", constant_values=-1)
+
+        # Convert to original data type
+        patched_image, *_ = convert_to_dst_type(src=patched_image, dst=image)
+        if self.return_location:
+            locations, *_ = convert_to_dst_type(src=locations, dst=image)
+
+        return patched_image, locations
+
+    def create_patches(self, input_image):
```

Comment:
Could you please help add doc-string and typehint for all these newly added member methods?

Thanks.